devices/src/gic.rs

```diff
@@ -54,12 +57,23 @@ impl Gic {
             .create_vgic(Gic::create_default_config(vcpu_count as u64))
             .map_err(Error::CreateGic)?;
 
-        let gic = Gic {
+        let mut gic = Gic {
             interrupt_source_group,
-            vgic: Some(vgic),
+            vgic: Some(vgic.clone()),
         };
         gic.enable()?;
 
+        if let Some(saved_vcpu_states) = saved_vcpu_states {
```

Comment:
In a temporary commit https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4960/commits/29b3a0fb08bf8be2857aede2ae47dad6e3df2586, I moved these code out of this function, but immediately after this function. Then the `Unexpected VM exit on "immediate_exit" run` error was gone, and the restoration succeeded. 

I don't understand why this happen.